### PR TITLE
LOG-3447: fix logging-view-plugin deployment to enable features based…

### PR DESCRIPTION
**Pull request by jcantrill**
… upon OCP version

### Description
This PR:
* Enables logging-view-plugin features based upon the version of OCP

### Links
* https://issues.redhat.com/browse/LOG-3447

cc @alanconway @jgbernalp

### CODE REVIEW

File: internal/consoleplugin/reconciler.go [hunk 1]
```diff
@@ -229,6 +230,8 @@ func (r *Reconciler) mutateDeployment() error {
 								Protocol:      "TCP",
 							},
 						},
+						Command: []string{"plugin-backend"},
+						Args:    []string{"-port", "9002", "-features", strings.Join(r.Config.Features, ",")},
```
Comment (jgbernalp) [inline internal/consoleplugin/reconciler.go:234]: Nginx was replaced with the go backend. So here we need to add the certificate and key args that the new backend supports: https://github.com/openshift/logging-view-plugin/pull/55/commits/a39036acf1dad1cd0ec28bdd6ef055697db5b1db#diff-4d25889b839b563a7b3536730d4aa8bf08b34a5fa25702dc3bfcd9dbfdca444cR11, also the port should be `9443`

File: internal/consoleplugin/config.go [hunk 2]
```diff
@@ -24,12 +25,13 @@ func (cf *Config) defaultMode() *int32 { return utils.GetInt32(420) }
 func (cf *Config) nginxPort() int32    { return 9443 }
```
Comment (jgbernalp) [inline internal/consoleplugin/config.go:25]:
```suggestion
func (cf *Config) pluginBackendPort() int32    { return 9443 }
```

**Review by alanconway — APPROVED** [review]
Looks good to me, will let @jcantrill give final /lgtm when he removes the /hold.

File: internal/consoleplugin/reconciler.go [hunk 3]
```diff
@@ -229,6 +230,13 @@ func (r *Reconciler) mutateDeployment() error {
...
 							},
 						},
+						Command: []string{"plugin-backend"},
+						Args: []string{
+							"-port", "9443",
+							"-features", strings.Join(r.Config.Features, ","),
+							"-cert", "/var/serving-cert/tls.crt",
+							"-key", "/var/serving-cert/tls.key",
+						},
 						VolumeMounts: []corev1.VolumeMount{
```
Comment (jgbernalp) [inline internal/consoleplugin/reconciler.go:240]: We probably don't need the nginx-config anymore

**Pull request merged**